### ISSUE TITLE
Module not found: Error: Can't resolve...

### ISSUE THREAD

**Issue by andryfailli** [nicojs/angular-stryker-example#1]
Hi, I've tried to use angular stryker in my project https://github.com/andryfailli/remindme-frontend/tree/feature/stryker following https://github.com/nicojs/angular-stryker-example#enable-mutation-testing-in-your-angular-project but I'm stuck with the following "Module not found" error about firebase (I'm using AngularFire2 & Firebase Auth).

Do you have an idea about what's wrong with my configuration?

Thank you

> [2018-01-29 11:48:53.576] [INFO] ConfigReader - Using stryker.conf.js in the current working directory.
> [2018-01-29 11:49:50.795] [INFO] TypescriptConfigEditor - Loading tsconfig file C:\Users\Andrea\git\remindme-frontend\tsconfig.json
> [2018-01-29 11:50:29.654] [INFO] InputFileResolver - Found 31 of 69 file(s) to be mutated.
> [2018-01-29 11:50:29.682] [INFO] InitialTestExecutor - Starting initial test run. This may take a while.
> [2018-01-29 11:55:03.060] [ERROR] StrykerCli - an error occurred Error: Could not transpile input files: Execute webpack: Error:
> ERROR in ./node_modules/angularfire2/firebase.app.module.js
> Module not found: Error: Can't resolve 'firebase/app' in 'C:\Users\Andrea\git\remindme-frontend\node_modules\angularfire2'
>  @ ./node_modules/angularfire2/firebase.app.module.js 2:0-41
>  @ ./node_modules/angularfire2/index.js
>  @ ./node_modules/angularfire2/auth/auth.module.js
>  @ ./node_modules/angularfire2/auth/public_api.js
>  @ ./node_modules/angularfire2/auth/index.js
>  @ ./src/app/auth-service/auth.service.ts
>  @ ./src/app/auth-guard/auth.guard.spec.ts
>  @ ./src \.spec\.ts$
>  @ ./src/test.ts
> 
> ERROR in ./src/app/auth-service/auth.service.ts
> Module not found: Error: Can't resolve 'firebase/app' in 'C:\Users\Andrea\git\remindme-frontend\src\app\auth-service'
>  @ ./src/app/auth-service/auth.service.ts 14:15-38
>  @ ./src/app/auth-guard/auth.guard.spec.ts
>  @ ./src \.spec\.ts$
>  @ ./src/test.ts
> 
> ERROR in ./src/app/auth-service/auth.service.spec.ts
> Module not found: Error: Can't resolve 'firebase/app' in 'C:\Users\Andrea\git\remindme-frontend\src\app\auth-service'
>  @ ./src/app/auth-service/auth.service.spec.ts 10:15-38
>  @ ./src \.spec\.ts$
>  @ ./src/test.ts
> 
> ERROR in ./node_modules/angularfire2/auth/auth.js
> Module not found: Error: Can't resolve 'firebase/auth' in 'C:\Users\Andrea\git\remindme-frontend\node_modules\angularfire2\auth'
>  @ ./node_modules/angularfire2/auth/auth.js 1:0-23
>  @ ./node_modules/angularfire2/auth/public_api.js
>  @ ./node_modules/angularfire2/auth/index.js
>  @ ./src/app/auth-service/auth.service.ts
>  @ ./src/app/auth-guard/auth.guard.spec.ts
>  @ ./src \.spec\.ts$
>  @ ./src/test.ts
> 
> ERROR in ./node_modules/angularfire2/auth/auth.module.js
> Module not found: Error: Can't resolve 'firebase/auth' in 'C:\Users\Andrea\git\remindme-frontend\node_modules\angularfire2\auth'
>  @ ./node_modules/angularfire2/auth/auth.module.js 2:0-23
>  @ ./node_modules/angularfire2/auth/public_api.js
>  @ ./node_modules/angularfire2/auth/index.js
>  @ ./src/app/auth-service/auth.service.ts
>  @ ./src/app/auth-guard/auth.guard.spec.ts
>  @ ./src \.spec\.ts$
>  @ ./src/test.ts
> 
> ERROR in ./src/app/auth-service/auth.service.ts
> Module not found: Error: Can't resolve 'firebase/messaging' in 'C:\Users\Andrea\git\remindme-frontend\src\app\auth-service'
>  @ ./src/app/auth-service/auth.service.ts 15:0-29
>  @ ./src/app/auth-guard/auth.guard.spec.ts
>  @ ./src \.spec\.ts$
>  @ ./src/test.ts
>     at InitialTestExecutor.<anonymous> (C:\Users\Andrea\git\remindme-frontend\node_modules\stryker\src\process\InitialTestExecutor.js:54:31)
>     at step (C:\Users\Andrea\git\remindme-frontend\node_modules\tslib\tslib.js:131:27)
>     at Object.next (C:\Users\Andrea\git\remindme-frontend\node_modules\tslib\tslib.js:112:57)
>     at fulfilled (C:\Users\Andrea\git\remindme-frontend\node_modules\tslib\tslib.js:102:62)
>     at process._tickCallback (internal/process/next_tick.js:109:7)

**Comment by nicojs** [comment]
Hi @andryfailli thanks for trying out Stryker with angular and reporting this issue. I can reproduce it, but need some more time to investigate what's going on. I will report back if i know more.

**Comment by andryfailli** [comment]
fantastic, thank you!

**Comment by nicojs** [comment]
I found out that the problem is with the InputFileSystem we've created for the in-memory compilation with webpack (see linked issue). I'll be working on a solution in the next couple of days, stay tuned.

Thanks for this issue, i tried this config on a number of different angular projects, but didn't come across this problem.

**Comment by nicojs** [comment]
@andryfailli I've created a PR that fixes the issue. I've tested it locally and the fix works on your project. See https://github.com/stryker-mutator/stryker/pull/601

Unfortunately there is another problem, namely that the "firebase-messaging-sw.js" file you have marked as an asset is also loaded into karma, resulting in this error:

```
Uncaught ReferenceError: importScripts is not defined
```

The mysterious way in which Stryker decides which files are to be loaded into the test runner (karma) is something that we are planning to improve in https://github.com/stryker-mutator/stryker/issues/589. In the meantime you can fix this yourself by removing "firebase-messaging-sw.js" from your assets manually when building your webpack config. Add this line in webpack-stryker.config.js on line 8:

```
appConfig.assets = appConfig.assets.filter(asset => asset !== 'firebase-messaging-sw.js');
```

**Comment by nicojs** [comment]
Fixed with stryker-webpack-transpiler@0.1.2

**Comment by andryfailli** [comment]
Hi @nicojs, that's fantastic, now it works like a charm. Thank you!
